### PR TITLE
Added support for advanced AwsClientParameters configuration (e.g. public S3 buckets) and tests

### PR DESCRIPTION
**Pull request by drfraser**
## Summary
Added config parameter to AwsClientParameters to support advanced configuration (e.g. accessing public S3 buckets)

## Relevant Issue(s)
Issues #104 and #102

## Checklist
- [x] Summarized PR's changes in the **Unreleased** section of the [CHANGELOG.md](https://github.com/PrefectHQ/prefect-aws/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by drfraser** [comment]
Note: I tried to make as few changes to the tests as possible and one is explicitly set as "expected to fail" (i.e. non-public S3 bucket and downloading a file w/o right configuration) - not sure of Prefect's testing philosophy.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Fix configuration to submit doc edits via GitHub - [#110](https://github.com/PrefectHQ/prefect-aws/pull/110)
 - Add `ECSTask.cloudwatch_logs_options` for customization of CloudWatch logging — [#116](https://github.com/PrefectHQ/prefect-aws/pull/116)
+- Added `config` parameter to AwsClientParameters to support advanced configuration (e.g. accessing public S3 buckets) [#104](https://github.com/PrefectHQ/prefect-aws/pull/117)
```
Comment (ahuang11) [inline CHANGELOG.md:12]:
```suggestion
- Added `config` parameter to AwsClientParameters to support advanced configuration (e.g. accessing public S3 buckets) [#117](https://github.com/PrefectHQ/prefect-aws/pull/117)
```

File: tests/test_client_parameters.py [hunk 2]
```diff
@@ -29,6 +29,20 @@
...
+        #         region_name="eu_west_1",
+        #         retries={"max_attempts": 10, "mode": "standard"}
+        #     )),
+        #     {
+        #         "config": {
+        #             "region_name": "eu_west_1",
+        #             "retries": {"max_attempts": 10, "mode": "standard"},
+        #         }
+        #     },
+        # ),
```
Comment (ahuang11) [inline tests/test_client_parameters.py:45]: I think this can be uncommented since it tests that you can pass a config kwarg.

**Review by ahuang11 — CHANGES_REQUESTED** [review]
Thanks for taking the initiative to add support for config! Made minor suggestions.

File: tests/test_client_parameters.py (hunk 2 above)
Comment (drfraser) [reply tests/test_client_parameters.py:45]: But the test ultimately fails, thus my explanatory comment - I thought adding in the necessary code to check equality of botocore.client.Config (it isn't a dict) to the dict would be kind of out of scope. I am not a pytest expert so maybe there is an easier way?  I added the whole comment to clearly state this simple test had not been overlooked given the other ones.

Or perhaps add some pytest code to mark the test as xfail?  Not sure what the Prefect testing philosophy is

File: CHANGELOG.md (hunk 1 above)
Comment (drfraser) [reply CHANGELOG.md]: Fixed

File: tests/test_client_parameters.py (hunk 2 above)
Comment (drfraser) [reply tests/test_client_parameters.py]: ok, I have enabled the test and set it to xfail

File: tests/test_client_parameters.py [hunk 3]
```diff
@@ -29,6 +30,21 @@
...
+                    retries={"max_attempts": 10, "mode": "standard"}
+                )
+            ),
+            {
+                "config": {
+                    "region_name": "eu_west_1",
+                    "retries": {"max_attempts": 10, "mode": "standard"},
+                },
+            },
+            marks=pytest.mark.xfail(reason="boto3 Config not directly equivalent to a dict"),
```
Comment (ahuang11) [inline tests/test_client_parameters.py:46]: Maybe instead of xfail, we can try put this in a separate test to test equality of botocore.client.Config

**Comment by ahuang11** [comment]
Thanks for updating. Sorry for the late reply about testing; I don't think it should xfail; if it's out of scope of the current testing function, I think we should create a new test for it.

Also the static analysis is failing on some linting. To fix, can you run `pre-commit install && pre-commit run --all`.

File: tests/test_client_parameters.py (hunk 3 above)
Comment (drfraser) [reply tests/test_client_parameters.py]: done

**Review by ahuang11 — APPROVED** [review]

**Comment by ahuang11** [comment]
Seriously appreciate your help!

**Pull request merged**